# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/wgpu_bindings/src/server.rs
# Commit: aef2a12d7819
# Full Hash: aef2a12d7819577e36f239d5be3a248d42e34929
# Author: Dzmitry Malyshau <dmalyshau@mozilla.com>
# Date: 2022-01-05 09:33:53
# Description:
#   Bug 1746312 - Return WebGPU errors from RenderBundle creation r=jimb
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D134986
# ==============================================================================

diff -r b0072bfee83c -r aef2a12d7819 gfx/wgpu_bindings/src/server.rs
--- a/gfx/wgpu_bindings/src/server.rs	Tue Jan 04 23:00:53 2022 +0000
+++ b/gfx/wgpu_bindings/src/server.rs	Tue Jan 04 23:04:08 2022 +0000
@@ -28,6 +28,10 @@
             e = source.source();
         }
 
+        self.init_str(string);
+    }
+
+    fn init_str(&mut self, string: String) {
         assert_ne!(self.capacity, 0);
         let length = if string.len() >= self.capacity {
             log::warn!(
@@ -339,6 +343,9 @@
                     error_buf.init(err);
                 }
             }
+            DeviceAction::Error(message) => {
+                error_buf.init_str(message);
+            }
         }
     }
 
